static/docs/commands-reference/push.md

```diff
@@ -53,7 +126,84 @@ Push all data file caches from the current Git branch to the default remote:
 ```
 
 Push outputs of a specific dvc file:
+
 ```dvc
     $ dvc push data.zip.dvc
+
     [#################################] 100% data.zip
 ```
+
+## Examples: With dependencies
+
+Demonstrating the `--with-deps` flag requires a larger example.  First, assume
+a pipeline has been setup with these stages:
+
+```dvc
+    $ dvc pipeline show
+
+    data/Posts.xml.zip.dvc
+    Posts.xml.dvc
+    Posts.tsv.dvc
+    Posts-test.tsv.dvc
+    matrix-train.p.dvc
+    model.p.dvc
+    Dvcfile
+```
+
+The local cache has been modified such that the data files in some of these
+stages should be uploaded to the remote cache.
+
+```dvc
+   $ dvc status --cloud
+
+    	new:            data/model.p
+	    new:            data/matrix-test.p
+	    new:            data/matrix-train.p
+```
+
+One could do a simple `dvc push` to share all the data, but what if you only want
+to upload part of the data?
+
+```dvc
+    $ dvc push --remote r1 --with-deps matrix-train.p.dvc
+
+    (1/2): [####################] 100% data/matrix-test.p data/matrix-test.p
+    (2/2): [####################] 100% data/matrix-train.p data/matrix-train.p
+
+    ... Do some work based on the partial update
+
+    $ dvc push --remote r1 --with-deps model.p.dvc
+
+    (1/1): [####################] 100% data/model.p data/model.p
+
+    ... Pull the rest of the data
+
+    $ dvc push --remote r1
+
+    Everything is up to date.
+
+    $ dvc status --cloud
+
+    Pipeline is up to date. Nothing to reproduce.
+```
+
+With the first `dvc push` we specified a stage in the middle of the pipeline
+while using `--with-deps`.  This started with the named stage and searched
+backwards through the pipeline for data files to upload.  Because the stage
```

Comment:
Can we do a single space everywhere? :) (btw you have different styles in this document).

Reply:
Does anyone read the raw markdown?  What people are reading is the rendered markdown as HTML on the website, or else in the github repository.  The raw markdown is for editing.  Once it is rendered the number of spaces after a period, how lines are wrapped into paragraphs, and so on, all that is disappeared into the rendered HTML.

Reply:
I consider writing docs (at least with Markdown or Latex) the same as coding. It's easier for everyone to write/code when there is a common style guide. In this specific case - there are some editors that automatically remove extra spaces (especially trailing). So, if it happens that someone creates a PR to fix a simple mistake there are chances we end up with a lot of unnecessary changes all over the file.

Reply:
Good points, and I'll keep this in mind.  On the other hand consider how documentation editing is different than code editing.  Inserting a few words in the middle of a paragraph causes the rest of the paragraph to reflow -- if one is manually adjusting the text to fit into 80 characters per line.  Meaning, the one line with inserted words will overflow 80 columns, then every following line in that paragraph probably also overflows, resulting in an excessive diff.

For DVC docs I'm making sure to remove trailing spaces and to fit things into 80 columns.  And I've adjusted the settings to insert spaces rather than tabs when hitting TAB.